odoo/tools/query.py

```diff
@@ -272,3 +286,39 @@ def __len__(self) -> int:
 
     def __iter__(self) -> Iterator[int]:
         return iter(self.get_result_ids())
+
+    @property
+    def _(self) -> QueryBuilder:
+        # Mapping could be saved on the registry
+        model = next(model._name for model in self._env.registry.values() if model._table == self.table)
+        return QueryBuilder(self, self.table, model, 'id')
+
+
+class QueryBuilder(SQLable):
```

Comment:
The name is clearly overstated.
It's actually not a query builder, but an identifier builder.
That being said, such an object could replace all the "table aliases" we use in the ORM's implementation...